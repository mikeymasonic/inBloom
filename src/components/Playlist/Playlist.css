.Playlist{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  margin: 0 0 0 1em;
  height: 5.8em;
  width: 10em;
  border: 2px groove black;
  box-shadow: 0px 0px 3px 0px black;
  border-radius: 10px;
  background-color: white;
  color: black;
  padding: 0 0.3em 0 0.3em;

  a {
    /* border-radius: 5px; */
    border: 1px solid black;
    padding: 0.1em;
    margin: 0.1em;
    width: 95%;
    text-align: center;
    font-size: 0.8em;
  }

  a:hover {
    box-shadow: 0px 0px 3px 0px black;
  }

  a:active {
    transform: scale(0.9);
  }

  .title {
    font-weight: bold;
    padding: 0.2em 0 0.4em 0;
    font-size: 1.1em;
  }

  .scroll {
    width: 100%;
    margin-bottom: 0.3em;
  }

  .content {
    display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  }
}