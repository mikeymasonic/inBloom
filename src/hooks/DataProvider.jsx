import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const DataContext = createContext();

export const DataProvider = ({ children }) => {
  // const [song, setSong] = useState('../mp3/example2.mp3');
  const [song, setSong] = useState(null);
  const [mapLocation, setMapLocation] = useState(null);
  const [playerVisible, setPlayerVisible] = useState(false);
  const [visualizerDisplay, setVisualizerDisplay] = useState(false);

  const handleSong = (track) => {
    setSong(track);
    setMapLocation(track.mapLocation);
  };

  const handleMapLocation = (location) => {
    setMapLocation(location);
  };

  const handlePlayerVisible = (bool) => {
    setPlayerVisible(bool);
  };

  const handleVisualizerDisplay = (bool) => {
    setVisualizerDisplay(bool);
  };

  return (
    <DataContext.Provider
      value={{
        song,
        mapLocation,
        playerVisible,
        visualizerDisplay,
        handleSong,
        handleMapLocation,
        handlePlayerVisible,
        handleVisualizerDisplay,
      }}
    >
      {children}
    </DataContext.Provider>
  );
};

DataProvider.propTypes = {
  children: PropTypes.node,
};

// get state
export const useSong = () => {
  const { song } = useContext(DataContext);
  return song;
};

export const useMapLocation = () => {
  const { mapLocation } = useContext(DataContext);
  return mapLocation;
};

export const usePlayerVisible = () => {
  const { playerVisible } = useContext(DataContext);
  return playerVisible;
};

export const useVisualizerDisplay = () => {
  const { visualizerDisplay } = useContext(DataContext);
  return visualizerDisplay;
};

// get handlers
export const useHandleSong = () => {
  const { handleSong } = useContext(DataContext);
  return handleSong;
};

export const useHandleMapLocation = () => {
  const { handleMapLocation } = useContext(DataContext);
  return handleMapLocation;
};

export const useHandlePlayerVisible = () => {
  const { handlePlayerVisible } = useContext(DataContext);
  return handlePlayerVisible;
};

export const useHandleVisualizerDisplay = () => {
  const { handleVisualizerDisplay } = useContext(DataContext);
  return handleVisualizerDisplay;
};

