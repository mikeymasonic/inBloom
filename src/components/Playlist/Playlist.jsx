import React from 'react';
import { useHandleSong, useMapLocation } from '../../hooks/DataProvider';
import {
  songs,
} from '../../utils/data';
import styles from './Playlist.css';

const Playlist = () => {

  const mapLocation = useMapLocation();
  const handleSong = useHandleSong();
  // const song = useSong();

  // if (mapLocation === song.mapLocation[0])
  const playlistNodes = songs.map((song) => {
    // return <p key={song.title}>hi</p>;
    if (mapLocation === song.mapLocation) {
      // handleSong(songs[0]);
      return <a key={song.title} onClick={() => handleSong(song)}>{song.title}</a>;
    }
    // handleSong(songs[1]);
  });

  return (
    <section className={styles.Playlist}>
      {playlistNodes}
    </section>
  );
};

export default Playlist;
