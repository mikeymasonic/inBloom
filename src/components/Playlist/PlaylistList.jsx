import React from 'react';
import PropTypes from 'prop-types';
import { useHandleSong } from '../../hooks/DataProvider';
// import Modal from 'react-modal';
// import AwesomeSlider from 'react-awesome-slider';
// import AwesomeSliderStyles from '../../styles/styles.scss';
import {
  songs,
} from '../../utils/data';
import styles from './PlaylistList.css';


const PlaylistList = () => {
  const handleSong = useHandleSong();

  const songNodes = songs.mapLocations.map((song) => {
    return <a key={song.title} onClick={() => handleSong(song)}>{song.title}</a>;
  });

  return (
    <section className={styles.PlaylistList}>
      {...songNodes}
    </section>
  );
};

PlaylistList.propTypes = {
  title: PropTypes.string.isRequired,
};

export default PlaylistList;
