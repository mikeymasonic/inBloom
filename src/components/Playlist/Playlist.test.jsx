import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { DataProvider } from '../../hooks/DataProvider';
import Playlist from './Playlist';

describe('Player component', () => {
  afterEach(() => cleanup());
  it('renders Player', () => {
    const { asFragment } = render(
      <Router>
        <DataProvider>
          <Playlist />
        </DataProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
