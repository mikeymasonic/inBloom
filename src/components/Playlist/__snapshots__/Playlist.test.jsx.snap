// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component renders Player 1`] = `
<DocumentFragment>
  <section
    class="Playlist"
    style="visibility: hidden; height: 0px;"
  >
    <div
      class="scrollarea scroll"
    >
      <div
        class="scrollarea-content content"
        style="margin-top: 0px; margin-left: 0px;"
        tabindex="1"
      />
    </div>
  </section>
</DocumentFragment>
`;
